src/commands/demo-store.js

```diff
@@ -0,0 +1,361 @@
+      return this.stores
+    }
+
+    const spinner = ora({
+      text: 'Loading example stores...',
+      stream: process.stdout,
+    }).start()
+
+    const manifest = await got('https://raw.githubusercontent.com/chec/example-stores/master/manifest.json').then(
+      response => JSON.parse(response.body)
+    )
+
+    spinner.stop()
+
+    this.cache.set({storeManifest: manifest})
+    this.stores = manifest
+
+    return this.stores
+  }
+
+  /**
+   * Ask the user to specify a store from those that are available
+   *
+   * @return {Promise<string>} A promise that resolves to the slug for the chosen store
+   */
+  async askForStore() {
+    // Grab the stores and _clone_ the array
+    const stores = (await this.retreiveStores()).slice(0)
+
+    // Return the only store if there's one
+    if (stores.length === 1) {
+      return stores.pop().slug
+    }
+
+    // Prep a closure to shorten the demo store description
+    const wrapString = string => {
+      const singleLine = string.replace(/[\n\r]+/g, ' ')
+      if (singleLine.length > 40) {
+        return `${singleLine.substring(0, 37)}...`
+      }
+      return singleLine
+    }
+
+    const {slug} = await inquirer.prompt([{
+      type: 'list',
+      name: 'slug',
+      message: 'Please choose a demo store to install:',
+      choices: stores.map(store => ({
+        name: `${chalk.bold(store.name)} ${chalk.dim(`(${store.slug})`)}: ${wrapString(store.description || '')}`,
+        value: store.slug,
+      })),
+    }])
+
+    return slug
+  }
+
+  /**
+   * Get the destination directory where the example project should be downloaded to. This will cause the command to
+   * exit if the calculated destination direction already exists and is not empty
+   *
+   * @param {object} manifest The "manifest" for the store that is being downloaded
+   * @return {string} The directory that the example should be downloaded to
+   */
+  getDestinationDirectory(manifest) {
+    // Run this only once
+    if (this.destinationDirectory) {
+      return this.destinationDirectory
+    }
+
+    // Allow specifying a target directory before defaulting to the stores slug
+    const {args: {targetDirectory}} = this.parse(DemoStoreCommand)
+
+    this.destinationDirectory = targetDirectory && targetDirectory.startsWith(sep) ?
+      targetDirectory : `${process.cwd()}${sep}${targetDirectory || manifest.slug}`
+
+    // Check if the folder exists
+    try {
+      fs.accessSync(this.destinationDirectory)
+    } catch (error) {
+      if (error.code === 'ENOENT') {
+        // Folder doesn't exist which is fine
+        return this.destinationDirectory
+      }
+      throw error
+    }
+
+    // Check that if the directory exists, it's empty
+    if (fs.readdirSync(this.destinationDirectory).length > 0) {
+      this.log(chalk.yellow(`The destination directory (${chalk.white.dim(
+        this.destinationDirectory
+      )}) already exists and is not empty`))
+      this.exit(1)
```

Comment:
Two comments here:

* I hit this after logging out and running demo-store to see how it handles unauthenticated users. Now it won't build again. I think the bug there is that the command shouldn't initialise the directory if you're not logged in.
* The "target directory" argument doesn't have a prompt, perhaps it should? That way I can tell it to install somewhere other than the default path

Reply:
I didn't do a prompt for a directory as you don't get auto-completion for a prompt. I was using tools like composer as a reference as well, which will choose a default for you. I understand the annoyance from the first issue but it's pretty trivial to `rm -rf` I thought?

Reply:
Of course, I'm just considering UX here. We'll be promoting this command as a way to try Commerce.js out in the docs, so it's probably quite common for someone to forget to login, perhaps especially if they're used to create-react-app which doesn't need any auth first (a one line install) - may not be worth doing though

Reply:
I'm moving the login check up in the code. I'll consider this resolved by that?

Reply:
Yep. I’m adding the check higher up in a separate branch now anyway